controllers/baremetalmachine_controller.go

```
@@ -94,7 +103,7 @@ func (r *BareMetalMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result,
 	// Fetch the Cluster.
 	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, capiMachine.ObjectMeta)
 	if err != nil {
-		er := errors.Wrapf(err, "BareMetalMachine's owner Machine is missing cluster label or cluster does not exist")
```

By doing this you are hiding the actual error, that will make debugging harder since the error won't appear in the logs

---

This pattern is used though the code. We need to have a convention on what style to use.

---

actually no, the New is only used when we do not have an error already. When we do, we just wrap

---

removed the wrapping from the `return`